gammapy/maps/core.py

```diff
@@ -887,8 +887,16 @@ def interp_to_geom(self, geom, preserve_counts=False, fill_value=0, **kwargs):
             data = map_copy.get_by_coord(coords)
             data = np.nan_to_num(data, nan=fill_value).astype(bool)
         else:
-            data = map_copy.interp_by_coord(coords, **kwargs)
-
+            data = map_copy.interp_by_coord(coords, fill_value=None, **kwargs)
+            if fill_value is not None: 
+                 idxs = self.geom.coord_to_idx(coords, clip=False) 
+                 invalid = np.broadcast_arrays(*[idx == -1 for idx in idxs]) 
```

Comment:
I think it is a good idea to introduce something like a `WcsGeom.footprint_mask` (lazy) attribute. I remember we have many of these checks, like `[idx == -1 for idx in idxs]`.It's cleaner and more efficient to just compute it once in a single place. Because it I snot going to change for a given `WcsGeom`.

Reply:
but here `idxs` depends on the `coords` of the target geom here so we have to recompute this

Reply:
But you can still extract the value of the pixels of the target geom like `self.geom.footprint_mask.get_by_coord()`?

Reply:
`get_by_coord ` will call `contains` so it would be simpler to do direclty:
`invalid = ~self.geom.contains(coords)`

Reply:
`contains` should use `geom.footprint_mask` as well. Let's do it like this: I'll introduce `footprint_mask` in a separate PR myself. There are easily ~5 more places where it could be used. Then you can rebase this PR and use it as well...